server/auth/session.guard.ts

```diff
@@ -42,16 +42,39 @@ export class SessionGuard implements CanActivate {
                     role: session?.identity?.traits?.role,
                     status: session?.identity.state,
                 };
-                return true;
+                const overridePublicRoutes =
+                    session?.identity?.traits?.role === "regular" &&
+                    this.configService.get<string>("override_public_routes");
+
+                if (overridePublicRoutes) {
+                    return this.checkRole(request, response, isPublic);
+                } else return true;
             }
 
             return this.next(request, response, isPublic);
         } catch (e) {
-            // TODO: logging
             return this.next(request, response, isPublic);
         }
     }
 
+    checkRole(request, response, isPublic) {
```

Comment:
suggestion: most of the logic is also used in the `next()` function, maybe you should consolidate them into a single function.